src/MotionControl.c

```diff
@@ -1355,79 +1355,76 @@ static STATUS Ros_Controller_DisableEcoMode()
 //
 // NOTE: only attempts to start job if necessary, does not reset errors, alarms.
 //       Does attempt to enable servo power (if not on)
-//       Does attempt to set the cycle mode to continuous (if not set)
+//       Does attempt to set the cycle mode to AUTO (if not set)
 //-----------------------------------------------------------------------
-BOOL Ros_MotionControl_StartMotionMode(MOTION_MODE mode)
+MotionNotReadyCode Ros_MotionControl_StartMotionMode(MOTION_MODE mode, rosidl_runtime_c__String* responseMessage)
 {
     int ret;
     MP_STD_RSP_DATA rData;
     MP_START_JOB_SEND_DATA sStartData;
-    int checkCount;
-    int grpNo;
+    int checkCount, grpNo, alarmcode;
+    MotionNotReadyCode motion_readiness_code;
+    char output[MOTION_START_ERROR_MESSSAGE_LENGTH] = { 0 };
 
     Ros_Debug_BroadcastMsg("%s: enter", __func__);
 
     if (Ros_MotionControl_ActiveMotionMode != MOTION_MODE_INACTIVE &&
         Ros_MotionControl_ActiveMotionMode != mode)
     {
         Ros_Debug_BroadcastMsg("Another trajectory mode (%d) is already active.", mode);
-        return FALSE;
+        return MOTION_NOT_READY_OTHER_TRAJ_MODE_ACTIVE;
     }
 
-    // Update status
     Ros_Controller_IoStatusUpdate();
-
     // Check if already in the proper mode
-    if(Ros_Controller_IsMotionReady())
+    if (Ros_Controller_IsMotionReady())
     {
         Ros_Debug_BroadcastMsg("Already active");
-        return TRUE;
+        return MOTION_READY;
     }
+    motion_readiness_code = Ros_Controller_GetNotReadySubcode(true);
 
-#ifndef DUMMY_SERVO_MODE
-    // Check for condition that need operator manual intervention
-    if(!Ros_Controller_IsRemote())
-    {
-        Ros_Debug_BroadcastMsg("Not remote, can't enable trajectory mode");
-        return FALSE;
-    }
-#endif
-
-    if (Ros_Controller_IsAnyFaultActive())
+    //Return with code if any of the problems are intractable
+    switch (motion_readiness_code)
     {
-        Ros_Debug_BroadcastMsg("Controller is in a fault state. Please call /reset_error");
-        return FALSE;
-    }
-
-    // Check if currently in operation, we don't want to interrupt current operation
-    if (Ros_Controller_IsOperating())
-    {
-        if (!Ros_Controller_MasterTaskIsJobName(g_nodeConfigSettings.inform_job_name))
-        {
+        case MOTION_NOT_READY_ESTOP:
+        case MOTION_NOT_READY_NOT_PLAY:
+        case MOTION_NOT_READY_NOT_REMOTE:
+        case MOTION_NOT_READY_HOLD:
+            return motion_readiness_code;
+        case MOTION_NOT_READY_ERROR:
+            alarmcode = Ros_Controller_GetAlarmCode();
+            snprintf(output, MOTION_START_ERROR_MESSSAGE_LENGTH, "%s: '%s' (0x%04X)",
```

Comment:
I don't understand why this is the only case where the return string gets populated

Reply:
Ok. Just add some comment that states the message is being populated here because that this location has the most information about the error. Also, that other cases will populate the message upstream.